### ISSUE TITLE
CRASH (linux loading private library)

### ISSUE THREAD

**Issue by derekbruening** [DynamoRIO/dynamorio#489]
_From [bruen...@google.com](https://code.google.com/u/109494838902877177630/) on June 07, 2011 14:45:23_

split from issue #480 comment 4:

on Ubuntu 10.04 w/ the private loader we hit a SIGSEGV here:

#0  0x00007f0bc1200b28 in __strcasecmp (s1=0x72000f43 "libstdc++.so.6", s2=0x2fe229a92f <Address 0x2fe229a92f out of bounds>) at strcasecmp.c:65
#1  0x00000000711dd1bf in privload_lookup (name=0x72000f43 "libstdc++.so.6") at /home/username/DR/core/loader_shared.c:273
#2  0x00000000712a15a8 in privload_process_imports (mod=0x4157fb38) at /home/username/DR/core/linux/loader.c:462
#3  0x00000000711ddd2e in privload_load_finalize (privmod=0x4157fb38) at /home/username/DR/core/loader_shared.c:499
#4  0x00000000711dcddc in loader_init () at /home/username/DR/core/loader_shared.c:119
#5  0x00000000710805ab in dynamorio_app_init () at /home/username/DR/core/dynamo.c:496
#6  0x00007f0bc20d0b00 in _init () at /home/username/DR/core/linux/preload.c:186
#7  0x00007f0bc22e5d25 in call_init (main_map=0x7f0bc24f8128, argc=1, argv=0x7fffe229ba78, env=0x7fffe229ba88) at dl-init.c:70
#8  _dl_init (main_map=0x7f0bc24f8128, argc=1, argv=0x7fffe229ba78, env=0x7fffe229ba88) at dl-init.c:134
#9  0x00007f0bc22d7b2a in _dl_start_user () from /lib64/ld-linux-x86-64.so.2
#10 0x0000000000000001 in ?? ()

basically we've had to revert to pre-private-loader to get this C++ client to work

_Original issue: http://code.google.com/p/dynamorio/issues/detail?id=489_
